controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -278,13 +278,21 @@ func (r *DataScienceClusterReconciler) reconcileSubComponent(ctx context.Context
 	componentName := component.GetComponentName()
 
 	enabled := component.GetManagementState() == v1.Managed
+
+	isExistStatus := false
+	_, isExistStatus = instance.Status.InstalledComponents[componentName]
+
 	// First set conditions to reflect a component is about to be reconciled
 	instance, err := status.UpdateWithRetry(ctx, r.Client, instance, func(saved *dsc.DataScienceCluster) {
 		message := "Component is disabled"
 		if enabled {
 			message = "Component is enabled"
 		}
-		status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileInit, message, corev1.ConditionUnknown)
+
+		// only set to init condition e.g Unknonw for the very first time when component is not in the list
+		if !isExistStatus {
```

Comment:
Looks like the whole update is under the condition

Reply:
yes, the idea is to, only make this call, if we find the component in the installcomponents list,
it indicates if this is happening after upgrade operator or it is another new reconcile. so we can skip the Unknown showing after it is True or False

Reply:
Yeah, but should you skip UpdateWithRetry() call in this case at all?

Reply:
ah, i see your point.